src/uv_tcp.c

```diff
@@ -46,13 +40,12 @@ void UvTcpMaybeFireCloseCb(struct UvTcp *t)
         return;
     }
 
-    assert(QUEUE_IS_EMPTY(&t->accepting));
-    assert(QUEUE_IS_EMPTY(&t->connecting));
```

Comment:
The removal of these assertions looks odd. The `UvTcpMaybeFireCloseCb` is meant to be called only after  `uvTcpClose` is called, which in turn calls `UvTcpListenClose` and `UvTcpConnectClose`. At that point all incoming/outgoing connections in the `accepting`/`connecting` queues should have been moved to the `aborting` queue.

If that's not the case anymore we should understand why.

Reply:
> The first assertion only got moved down a few files and is still in place. The later one got replaced by extending the if condition a few lines below. I will explain the need for this replacement in my response a few lines below.

Is the reason the addition of the `vTcpMaybeFireCloseCb(t)` call at then end of `UvTcpListenClose()`? Like I mentioned in the other comment. I'd expect to be just that, or is there more?

Reply:
Moved the UvTcpMaybeFireCloseCb call at the end of the uvTcpClose instead now. And this way it works fine now. This way I could put back in the assertions into the UvTcpMaybeFireCloseCb as well.